### PR TITLE
test: begin implementing basic component tests

### LINKED ISSUE #112: Add more complete test coverage
Just to make sure we don't break things 👍

### PR DESCRIPTION
**Pull request by andrewda**
Re-opening #148 as a branch of this main repo so it's easier for people to contribute to these tests while they're being developed.

Fixes #112.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -54,24 +56,25 @@
...
-    "react-native-parallax-scroll-view": "^0.19.0",
+    "react-native-mock": "https://github.com/shqld/react-native-mock/tarball/master",
+    "react-native-parallax-scroll-view": "https://github.com/mini-eggs/react-native-parallax-scroll-view/tarball/react16-support",
     "react-native-photo-view": "^1.5.2",
     "react-native-safari-view": "^2.0.0",
-    "react-native-search-bar": "^3.0.0",
+    "react-native-search-bar": "https://github.com/andrewda/react-native-search-bar/tarball/master",
     "react-native-syntax-highlighter": "^1.2.1",
-    "react-native-table-component": "^1.0.5",
+    "react-native-table-component": "https://github.com/jaynakus/react-native-table-component/tarball/master",
```
Comment (andrewda) [inline package.json]: Can be set to `^1.1.0` :tada:

Comment (alejandronanez) [reply package.json]: Why is this pointing to github instead of a regular version number?

Comment (andrewda) [reply package.json]: This is the fork that works with React 16. Now 1.1.0 works, too

Comment (alejandronanez) [reply package.json]: Oh, makes sense! Thanks for clarification @andrewda!

**Comment by andrewda** [comment]
I think #493 just needs to me merged, then this will be GTG.

**Comment by jouderianjr** [comment]
I would love to help the implementation of the tests \o/ Once on master, how could we split the job? Divide and conquer 💃

**Comment by alejandronanez** [comment]
We should do the same as we did for styled-components, create different issues so people can start contributing one component at a time. 😎

File: __tests__/tests/components/ToggleView.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import sinon from 'sinon';
+import { shallow, render } from 'enzyme';
+
+import { ToggleView } from 'components';
+
+describe('<ToggleView />', () => {
+  it('correctly renders with default touchable view', () => {
+    const wrapper = render(
+      <ToggleView>
+        <Text>I'm hidden</Text>
```
Comment (andrewda) [inline __tests__/tests/components/ToggleView.js]: GitHub doesn't like these quotation marks for some reason (though I believe they'll compile fine). Maybe just replace with "I am hidden"?

File: __tests__/tests/components/ToggleView.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        .find('text')
+        .first()
+        .text()
+    ).toEqual('…');
+  });
+
+  it('correctly renders with a custom touchable view', () => {
+    const wrapper = render(
+      <ToggleView TouchableView={<Text>CUSTOM</Text>}>
+        <Text>I'm hidden</Text>
```
Comment (andrewda) [inline __tests__/tests/components/ToggleView.js]: ^

File: __tests__/tests/components/ToggleView.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        .find('text')
+        .first()
+        .text()
+    ).toEqual('CUSTOM');
+  });
+
+  it('starts with the hidden element collapsed', () => {
+    const wrapper = shallow(
+      <ToggleView TouchableView={<Text>CUSTOM</Text>}>
+        <Text>I'm hidden</Text>
```
Comment (andrewda) [inline __tests__/tests/components/ToggleView.js]: ^

File: __tests__/tests/components/ToggleView.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    );
+
+    expect(wrapper.find('Collapsible').prop('collapsed')).toEqual(true);
+  });
+
+  it('uncollapses the hidden element when touched', () => {
+    const onPress = sinon.spy();
+    const wrapper = shallow(
+      <ToggleView>
+        <Text>I'm hidden</Text>
```
Comment (andrewda) [inline __tests__/tests/components/ToggleView.js]: ^

File: __tests__/tests/components/ToggleView.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    wrapper.find('TouchableOpacity').simulate('press');
+
+    expect(wrapper.find('Collapsible').prop('collapsed')).toEqual(false);
+  });
+
+  it('recollapses the hidden element when touched twice', () => {
+    const onPress = sinon.spy();
+    const wrapper = shallow(
+      <ToggleView>
+        <Text>I'm hidden</Text>
```
Comment (andrewda) [inline __tests__/tests/components/ToggleView.js]: ^

**Comment by andrewda** [comment]
I'll fix the comment I left above quickly, then I'd just like to get a few more confirmations that this works on Android, because this is making the significant leap to React v16.

**Comment by andrewda** [comment]
@jouderianjr @alejandronanez Check out #518 for dividing up the work!

**Pull request merged**